vllm/worker/worker_base.py

```diff
@@ -125,6 +125,8 @@ def update_environment_variables(envs: Dict[str, str]) -> None:
             # overwriting CUDA_VISIBLE_DEVICES is desired behavior
             # suppress the warning in `update_environment_variables`
             del os.environ[key]
+            if is_hip():
+                os.environ.pop("HIP_VISIBLE_DEVICES", None)
```

Comment:
Can we raise a warning if `HIP_VISIBLE_DEVICES` is not None?

Reply:
Have added a warning and some flavor text to emphasize that if both `CUDA_VISIBLE_DEVICES` and `HIP_VISIBLE_DEVICES` is set, `CUDA_VISIBLE_DEVICES` takes precedence. Hopefully we can revert the need for this approach once ROCm-PyTorch starts recognizing `CUDA_VISIBLE_DEVICES` correctly.